orderer/common/channelparticipation/restapi.go

```diff
@@ -50,23 +78,122 @@ func (h *HTTPHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	switch req.Method {
-	case http.MethodGet, http.MethodPost, http.MethodDelete:
-		//TODO skeleton
-		err := errors.Errorf("not implemented yet: %s %s", req.Method, req.URL.String())
-		h.sendResponseJsonError(resp, http.StatusNotImplemented, err)
+	h.router.ServeHTTP(resp, req)
+}
+
+// List all channels
+func (h *HTTPHandler) serveListAll(resp http.ResponseWriter, req *http.Request) {
+	_, err := negotiateContentType(req) // Only application/json for now
+	if err != nil {
+		h.sendResponseJsonError(resp, http.StatusNotAcceptable, err)
+		return
+	}
+	channelList := h.registrar.ListAllChannels()
+	for i, info := range channelList.Channels {
+		channelList.Channels[i].URL = path.Join(URLBaseV1Channels, info.Name)
+	}
+	h.sendResponseOK(resp, channelList)
+}
+
+// List a single channel
+func (h *HTTPHandler) serveListOne(resp http.ResponseWriter, req *http.Request) {
+	_, err := negotiateContentType(req) // Only application/json for now
+	if err != nil {
+		h.sendResponseJsonError(resp, http.StatusNotAcceptable, err)
+		return
+	}
 
-	default:
-		err := errors.Errorf("invalid request method: %s", req.Method)
-		h.sendResponseJsonError(resp, http.StatusBadRequest, err)
+	channelID, ok := mux.Vars(req)[channelIDKey]
+	if !ok {
+		h.sendResponseJsonError(resp, http.StatusInternalServerError, errors.Wrap(err, "missing channel ID"))
```

Comment:
should be ` http.StatusBadRequest` cause `http.StatusInternalServerError` means fatal error, while here is just an empty or missing channelID

Reply:
Actually this has nothing to do with the request, it is an indication of bug - that the key used in the pattern and Vars map (const channelIDKey) is inconsistent in registration & retrieval. That would be a repeatable error internal to the server that the client cannot correct by changing the request.

Reply:
I do not understand how missing channel id in the request path is internal server error. Can you elaborate?